src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -281,11 +284,68 @@ RecordPlacementAccessToCache(Oid relationId, ShardPlacementAccessType accessType
 }
 
 
+/*
+ * RecordParallelAccessForTaskList gets a task list and records
+ * the necessary parallel relation accesses for the task list.
+ *
+ * This function is used to enforce foreign keys from distributed
+ * tables to reference tables.
+ */
+void
+RecordParallelAccessForTaskList(List *taskList)
+{
```

Comment:
I think keeping the RecordParallel -> RecordRelationParallel convention would be nice. Then it's "obvious" why we only care about the first task.